src/Symfony/Component/Translation/Translator.php

```diff
@@ -266,21 +295,56 @@ public function getCatalogue($locale = null)
             $this->assertValidLocale($locale);
         }
 
-        if (!isset($this->catalogues[$locale])) {
+        if (isset($this->catalogues[$locale]) && $this->getMessageCatalogueProvider()->isFresh($locale)) {
+            return $this->catalogues[$locale];
+        }
+
+        // check if the Translator class is overrated
+        if ('Symfony\Component\Translation\Translator' !== get_class($this)) {
```

Comment:
Not sure if checking these en bloc here is a good thing. What about having these checks in those places where you're going to call one of the methods or in the constructor to make sure it's actually run? (What if `getCatalogue()` is overwritten itself?)